src/core/toga/fonts.py

```diff
@@ -34,3 +36,95 @@ def bind(self, factory):
 
     def measure(self, text, dpi, tight=False):
         return self._impl.measure(text, dpi=dpi, tight=tight)
+
+    @staticmethod
+    def register(family, path, weight=NORMAL, style=NORMAL, variant=NORMAL):
+        """
+        Registers a file-based font with it's family name, style, variant and weight.
+        When invalid values for style, variant or weight are passed, NORMAL will be used instead.
+
+        Platforms that support dynamic font loading will load the font from the supplied path when a toga style
+        references the registered font, e.g.
+
+        Font.register("Font Awesome 5 Free Solid", "resources/Font Awesome 5 Free-Solid-900.otf")
+
+        Font.register("Roboto", "resources/Roboto-Bold.ttf", weight=Font.BOLD)
+
+        Args:
+            family (str):  The font family name
+            path (str):    The path to the font file, relative to the application's module directory.
+            weight (str):  The font weight: Font.NORMAL (default) or a value from Font.FONT_WEIGHTS
+            style (str):   The font style: Font.NORMAL (default) or a value from Font.FONT_STYLES
+            variant (str): The font variant: Font.NORMAL (default) or a value from Font.FONT_VARIANTS
+        """
+        registered_font = RegisteredFont(
+            family, path, weight=weight, style=style, variant=variant
+        )
+        _REGISTERED_FONT_CACHE[registered_font.key] = registered_font
```

Comment:
If you define `__hash__()` on a class, you don't need to explicitly reference a key like this; you can just use the object as the key.

Reply:
I'm also a little unclear about what is going on here. You've got a dictionary mapping from a font... to a font. Why is the path *on* the RegisteredFont, rather than being the thing that is being registered?

Reply:
OK, I addressed this issue